src/main/scala/firrtl/ir/Serializer.scala

```diff
@@ -10,24 +10,29 @@ object Serializer {
   val NewLine = '\n'
   val Indent = "  "
 
-  /** Converts a `FirrtlNode` into its string representation. */
+  /** Converts a `FirrtlNode` into its string representation
+   * with *zero* indents (old behavior for binary compatibility) */
   def serialize(node: FirrtlNode): String = {
+    serialize(node, 0)
+  }
+
+  /** Converts a `FirrtlNode` into its string representation. */
+  def serialize(node: FirrtlNode, indents: Int): String = {
```

Comment:
```suggestion
  def serialize(node: FirrtlNode, indents: Int = 0): String = {
```
Can't we do away with the above version by just adding a default param? Does that still match binary compatibility? (I don't actually know the ansewr this is a weak suggestion)

Reply:
The default parameter approach was my first attempt at solving this, but I'm not sure how it would be compiled into bytecode. My intuition tells me that the serialize with the default parameter will still be a two-parameter function (with some fancy magic to "insert" the 0 as an argument). So if we swap the firrtl binary then anything else that tries to run that will get a complaint that the one-parameter serialize doesn't exist.

Reply:
looks like your intution is right: https://github.com/jatcwang/binary-compatibility-guide#dont-adding-parameters-with-default-values-to-methods

Not sure what the firrtl philosophy is on making this breaking change on master and then fixing in the backport version?

Reply:
I think just having 2 is fine.

I'm going to be super bikesheddy here though and suggest naming the parameter just `indent`:
```suggestion
  def serialize(node: FirrtlNode, indent: Int): String = {
```

Doing a bit of search on other projects, singular tense is more common for this kind of parameter.